hwilib/devices/trezor.py

```diff
@@ -95,6 +98,38 @@ def parse_multisig(script):
     multisig = proto.MultisigRedeemScriptType(m=m, signatures=[b''] * n, pubkeys=pubkeys)
     return (True, multisig)
 
+# Parses the PSBT_GLOBAL_XPUB fields of a PSBT as multisig pubkeys
+def parse_multisig_xpubs(tx, psbt_in_out, multisig):
+    try:
+        old_pubs = [k.node.public_key for k in multisig.pubkeys]
+        if len(old_pubs) != sum([len(xpubs) for xpubs in tx.global_xpubs.values()]):
+            # global xpubs weren't found
+            return multisig
+        new_pubs = []
+        xpubs = [item for sublist in tx.global_xpubs.values() for item in sublist]
+        derivations = [list(struct.unpack("<" + "I" * (len(der) // 4), der)) * len(tx.global_xpubs[der]) for der in tx.global_xpubs.keys()]
+        for pub in old_pubs:
+            der = list(psbt_in_out.hd_keypaths[pub])
+            for i, derivation in py_enumerate(derivations):
+                if der[0] == derivation[0]:
+                    idx = i
+                    for i in range(len(derivation)):
+                        if der[i] != derivation[i]:
+                            # derivations mismatch
+                            return multisig
+                    break
+            xpub = xpubs[idx]
```

Comment:
I think we would want to make sure that the xpub derives the pubkey we are looking at.